ui/client/components/graph/TypeSuggest.js

```diff
@@ -0,0 +1,134 @@
+import React from 'react'
+import PropTypes from "prop-types";
+import AceEditor from 'react-ace';
+import _ from "lodash";
+import ProcessUtils from "../../common/ProcessUtils";
+import ActionsUtils from "../../actions/ActionsUtils";
+import {connect} from 'react-redux'
+
+import 'brace/mode/jsx';
+
+import 'brace/ext/searchbox';
+import '../../brace/mode/spel'
+import '../../brace/mode/sql'
+import '../../brace/theme/nussknacker'
+import ValidationLabels from "../modals/ValidationLabels";
+import {allValid} from "../../common/Validators";
+import TypeSuggester from "./TypeSuggester";
+
+var inputExprIdCounter = 0;
+class TypeSuggest extends React.Component {
+
+  static propTypes = {
+    inputProps: PropTypes.object.isRequired,
+    fieldName: PropTypes.string,
+    validators: PropTypes.array.isRequired
+  }
+
+  constructor(props) {
+    super(props);
+    inputExprIdCounter+=1;
+    this.state = {
+      value: props.inputProps.value,
+      id: "inputExpr" + inputExprIdCounter
+    };
+    this.typeSuggester = this.createTypeSuggester(props)
+  }
+
+  createTypeSuggester = (props) => {
+    return new TypeSuggester(props.typesInformation);
+  }
+
+  onChange = (newValue) => {
+    this.setState({
+      value: newValue
+    })
+    this.props.inputProps.onValueChange({typ: {refClazzName: newValue}})
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    return !(_.isEqual(this.state.value, nextState.value))
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    this.typeSuggester = this.createTypeSuggester(this.props)
+
+    if (!_.isEqual(this.state.value, prevState.value)) {
+      this.props.inputProps.onValueChange({typ: {refClazzName: this.state.value}})
+    }
+  }
+
+  customAceEditorCompleter = {
+    getCompletions: (editor, session, caretPosition2d, prefix, callback) => {
+      const suggestions = this.typeSuggester.suggestionsFor(this.state.value, caretPosition2d)
+      callback(null, _.map(suggestions, (s) => {
+        const refClazzName = s.clazzName.refClazzName
+        const fieldType = ProcessUtils.humanReadableType(refClazzName)
+        return {
+          name: fieldType,
+          value: refClazzName,
+          meta: refClazzName,
+          returnType: refClazzName,
+          score: 1
+        }
+      }))
+    }
+  }
+
+  render() {
+    const {isMarked} = this.props
+    if (this.props.dataResolved) {
+      return (
+        <div>
+          <div style={{paddingTop: 8,
+                       paddingBottom: 8,
+                       paddingLeft: 10,
+                       paddingRight: 10,
```

Comment:
Shouldn't we move these inlined styles to css class?

Reply:
I agree, please move these to css class.

Reply:
done